src/TRoomDB.cpp

```diff
@@ -46,6 +46,23 @@ TRoomDB::TRoomDB(TMap* pMap)
     addArea(-1, mpMap->getDefaultAreaName());
 }
 
+TRoomDB::~TRoomDB()
+{
+    // Delete all rooms
+    QList<TRoom*> const rPtrL = getRoomPtrList();
+    for (auto room : rPtrL) {
+        delete room;
```

Comment:
:rotating_light: This will be slower than it needs to be because each `TRoom`  will be calling the very unwisely named (prefixing any identifier with a `_` is not recommended - and with a double `_` is actually a real no-no as they are [reserved](https://en.cppreference.com/w/cpp/language/identifiers)) `TRooomDB::__removeRoom(int)` which will, **unnecessarily for a complete map database destruction**, disconnect all the room's exits to other rooms and update some details of the area that it was in.

Reply:
A quick and dirty solution would be to set a private (to that class) flag in the `TRoomDB` destructor that when set causes `TRoom::__removeRoom(int)` to return without doing anything,,, not ideal as you still have the function call overhead but it should avoid most of the delays...